### ISSUE TITLE
OCS Inventory Agent not Synchronizing Plugins with Server

### ISSUE THREAD

**Issue by eliaspizarro** [OCSInventory-NG/OCSInventory-Server#62]
## General informations
Operating system Server : Debian GNU/Linux 7
Operating system Client : Windows 10 build 1703 x64
Apache Server version : 2.2.22

## Server informations
Perl version : 5.14.2
Mysql version :  5.5.55

## OCS Inventory informations
OCS server version : 2.3.1
OCS Agent version : 2.3

### Empty plugins data in OCS Server

DSKEPIZARRO XML
VIRTUAL MACHINES

Show  entries
Search : 
Name VM	Statut VM	Operating system VM	Type VM	UUID	Processor(s) VM	Memory VM	Computer	Operating system
No data available in table
Showing 0 to 0 of 0 entrie
PreviousNext
NETWORK SHARE

Show  entries
Search : 
Drive	Path
No data available in table
Showing 0 to 0 of 0 entrie
PreviousNext
TEAMVIEWER

Show  entries
Search : 
ID	Version
No data available in table
Showing 0 to 0 of 0 entrie
PreviousNext
WINDOWS UPDATE STATE

Show  entries
Search : 
Scheduled Install Date	Last Succes Time	Detect Succes Time	Download Succes Time
No data available in table

### If I push to local file from agent I get plugin data localy. For example:
...
```
<SECURITYCENTER>
	<SCV>2</SCV>
	<CATEGORY>AntiVirus</CATEGORY>
	<COMPANY>Microsoft Corporation</COMPANY>
	<PRODUCT>Windows Defender</PRODUCT>
	<VERSION>4.11.15063.0</VERSION>
	<ENABLED>0</ENABLED>
	<UPTODATE>0</UPTODATE>
</SECURITYCENTER>
<SECURITYCENTER>
	<SCV>2</SCV>
	<CATEGORY>AntiVirus</CATEGORY>
	<COMPANY>ESET</COMPANY>
	<PRODUCT>ESET Smart Security 9.0.408.1</PRODUCT>
	<VERSION>9.0.407.0</VERSION>
	<ENABLED>1</ENABLED>
	<UPTODATE>1</UPTODATE>
</SECURITYCENTER>
```
...

### Plugins into C:\Program Files (x86)\OCS Inventory Agent\Plugins with permissions
DO_NOT_REMOVE.txt Todos:(I)(RX,W)
                  ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
                  ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
                  NT AUTHORITY\SYSTEM:(I)(F)
                  BUILTIN\Administradores:(I)(F)
                  BUILTIN\Usuarios:(I)(RX)

msofficekey.vbs Todos:(I)(RX,W)
                ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
                ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
                NT AUTHORITY\SYSTEM:(I)(F)
                BUILTIN\Administradores:(I)(F)
                BUILTIN\Usuarios:(I)(RX)

networkshare.vbs Todos:(I)(RX,W)
                 ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
                 ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
                 NT AUTHORITY\SYSTEM:(I)(F)
                 BUILTIN\Administradores:(I)(F)
                 BUILTIN\Usuarios:(I)(RX)

security.vbs Todos:(I)(RX,W)
             ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
             ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
             NT AUTHORITY\SYSTEM:(I)(F)
             BUILTIN\Administradores:(I)(F)
             BUILTIN\Usuarios:(I)(RX)

teamviewer.vbs Todos:(I)(RX,W)
               ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
               ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
               NT AUTHORITY\SYSTEM:(I)(F)
               BUILTIN\Administradores:(I)(F)
               BUILTIN\Usuarios:(I)(RX)

winupdate.vbs Todos:(I)(RX,W)
              ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIONES:(I)(F)
              ENTIDAD DE PAQUETES DE APLICACIONES\TODOS LOS PAQUETES DE APLICACIÓN RESTRINGIDOS:(I)(F)
              NT AUTHORITY\SYSTEM:(I)(F)
              BUILTIN\Administradores:(I)(F)
              BUILTIN\Usuarios:(I)(RX)
			  
### The dafault data is pushed correctly from agent to server. For example:
...
```
<CONTROLLERS>
	<ID>80295</ID>
	<MANUFACTURER>Microsoft</MANUFACTURER>
	<NAME>Microsoft Storage Spaces Controller</NAME>
	<CAPTION>Microsoft Storage Spaces Controller</CAPTION>
	<DESCRIPTION>Microsoft Storage Spaces Controller</DESCRIPTION>
	<VERSION />
	<TYPE>SCSI Controller</TYPE>
</CONTROLLERS>
```
...

### Logs from agent
```
==============================================================================
Starting OCS Inventory NG Agent on Thursday, June 01, 2017 20:02:26.
AGENT => Running OCS Inventory NG Agent Version 2.3.0.0
AGENT => Using OCS Inventory NG FrameWork Version 2.3.0.0
AGENT => Loading plug-in(s)
	DLL PLUGIN => Searching for Plug-in DLL(s) in folder <C:\Program Files (x86)\OCS Inventory Agent\plugins>
	DLL PLUGIN => 0 DLL Plug-in(s) successfully loaded on 0 DLL(s) found
AGENT => Using network connection with Communication Server
	COM PROVIDER => Loading Communication Provider <C:\Program Files (x86)\OCS Inventory Agent\ComHTTP.dll>
AGENT => Using Communication Provider <OCS Inventory NG cURL Communication Provider> Version <2.3.0.0>
AGENT => Sending Prolog
	DID_CHECK => Read DeviceID <DSKEPIZARRO-2017-06-01-14-35-44> and MACs <******> in file <ocsinventory.dat>
	COM SERVER => Initializing cURL library for sendRequest
	COM SERVER => Using cURL without server authentication
	COM SERVER => Disabling cURL proxy support
	COM SERVER => Disabling cURL SSL server validation support
	COM SERVER => Sending HTTP Post request to URL <http://192.168.10.25/ocsinventory>
	COM SERVER => HTTP Post response received <HTTP Status Code #200>
	COM SERVER => Cleaning cURL library
================= TRACE START ===============
<?xml version='1.0' encoding='UTF-8'?>
<REPLY>
  <OPTION>
    <NAME>DOWNLOAD</NAME>
    <PARAM FRAG_LATENCY="10" PERIOD_LATENCY="1" ON="0" TYPE="CONF" CYCLE_LATENCY="60" TIMEOUT="30" PERIOD_LENGTH="10" EXECUTION_TIMEOUT="120" />
  </OPTION>
  <RESPONSE>SEND</RESPONSE>
  <INVENTORY_ON_STARTUP>1</INVENTORY_ON_STARTUP>
  <PROLOG_FREQ>24</PROLOG_FREQ>
</REPLY>

================= TRACE STOP ===============
AGENT => Prolog successfully sent
SUPPORT => No support detected, Registration key : N/A
	AGENT => Prolog Frequency set to 24 hour(s)
	AGENT => Inventory on startup option set to 1 
AGENT => Inventory required
AGENT => Launching hardware and software checks
	INVENTORY => Loading Download history
	INVENTORY => Logged on user ID is <******>
	INVENTORY => Operating System is <Microsoft Windows 10 Enterprise 10.0.15063 >, description <>
	INVENTORY => Operating System uses 64 bits memory address width
	INVENTORY => Computer domain or workgroup is <WORKGROUP>
	INVENTORY => User domain is <>
	INVENTORY => System Manufacturer <System manufacturer>, System Model <System Product Name>, System S/N <System Serial Number>, Bios Manufacturer <American Megatrends Inc.>, Bios Date <07/02/2013>, Bios Version <0504>, Motherboard Manufacturer <ASUSTeK COMPUTER INC.>, Motherboard Model <H61M-K>, Motherboard S/N <******>
	INVENTORY => 1 processor(s) Intel(R) Core(TM) i5-3570K CPU @ 3.40GHz [4 core(s) x86_64] at 3401 MHz
	INVENTORY => OS Memory 16331 MB, OS Swap size 32715 MB
	INVENTORY => 2 memory slot(s) found
	INVENTORY => 2 input device(s) found
	INVENTORY => 17 system port(s) found
	INVENTORY => 3 system slot(s) found
	INVENTORY => 5 system controller(s) found
	INVENTORY => 3 storage peripheral(s) found
	INVENTORY => 7 sound device(s) found
	INVENTORY => 1 modem(s) found
	INVENTORY => 5 network adapter(s) found
	INVENTORY => 2 system printer(s) found
	INVENTORY => 1 video adapter(s) found
	INVENTORY => 1 system monitor(s) found
	INVENTORY => Default IPv4 address is <******>
	INVENTORY => Registered company <>, registered owner <******>, Product ID <******-******...>
	INVENTORY => Product key <******-******...>
	INVENTORY => 428 software found
	INVENTORY => Computer/VM UUID is <******-*****-******-******-******>
	INVENTORY => Computer seems to be Microsoft virtual host
	INVENTORY => System Memory set to 16384 (instead of 16331)
	INVENTORY => Reading last inventory state
	EXECUTABLE PLUGIN => Searching for VBS script(s) in folder <C:\Program Files (x86)\OCS Inventory Agent\plugins>
	EXECUTABLE PLUGIN => Found executable plugin <msofficekey.vbs>
================= TRACE START ===============

================= TRACE STOP ===============
	EXECUTABLE PLUGIN => Plugin <msofficekey.vbs> executed successfully
	EXECUTABLE PLUGIN => Found executable plugin <networkshare.vbs>
================= TRACE START ===============

================= TRACE STOP ===============
	WARNING *** EXECUTABLE PLUGIN => Plugin <C:\Program Files (x86)\OCS Inventory Agent\plugins\networkshare.vbs> does not produce any output
	EXECUTABLE PLUGIN => Found executable plugin <security.vbs>
================= TRACE START ===============

================= TRACE STOP ===============
	EXECUTABLE PLUGIN => Plugin <security.vbs> executed successfully
	EXECUTABLE PLUGIN => Found executable plugin <teamviewer.vbs>
================= TRACE START ===============

================= TRACE STOP ===============
	EXECUTABLE PLUGIN => Plugin <teamviewer.vbs> executed successfully
	EXECUTABLE PLUGIN => Found executable plugin <winupdate.vbs>
================= TRACE START ===============

================= TRACE STOP ===============
	EXECUTABLE PLUGIN => Plugin <winupdate.vbs> executed successfully
	EXECUTABLE PLUGIN => Searching for executable(s) in folder <C:\Program Files (x86)\OCS Inventory Agent\plugins>
AGENT => Sending Inventory
	INVENTORY => Checking last inventory state
	INVENTORY => Logical drives inventory state changed
INVENTORY => Inventory changed since last run
	INVENTORY => Generating XML document with Device properties
	INVENTORY => XML Update BIOS
	INVENTORY => XML Update 1 CPU(s)
	INVENTORY => XML Update 2 Memory Slot(s)
	INVENTORY => XML Update 2 Input Device(s)
	INVENTORY => XML Update 17 System Port(s)
	INVENTORY => XML Update 5 System Controller(s)
	INVENTORY => XML Update 3 System Slot(s)
	INVENTORY => XML Update 7 Sound Device(s)
	INVENTORY => XML Update 3 Storage Peripheral(s)
	INVENTORY => XML Update 3 Logical Drive(s)
	INVENTORY => XML Update 1 Modem(s)
	INVENTORY => XML Update 5 Network Adapter(s)
	INVENTORY => XML Update 1 Video Adapter(s)
	INVENTORY => XML Update 1 Monitor(s)
	INVENTORY => XML Update 2 Printer(s)
	INVENTORY => XML Update 429 Software
	INVENTORY => XML Update 0 Registry Value(s)
	INVENTORY => XML Update Administrative Information(s)
	INVENTORY => XML Update common Device properties
	COM SERVER => Initializing cURL library for sendRequest
	COM SERVER => Using cURL without server authentication
	COM SERVER => Disabling cURL proxy support
	COM SERVER => Disabling cURL SSL server validation support
	COM SERVER => Sending HTTP Post request to URL <http://192.168.10.25/ocsinventory>
	COM SERVER => HTTP Post response received <HTTP Status Code #200>
	COM SERVER => Cleaning cURL library
================= TRACE START ===============
<?xml version='1.0' encoding='UTF-8'?>
<REPLY>
  <RESPONSE>NO_ACCOUNT_UPDATE</RESPONSE>
</REPLY>

================= TRACE STOP ===============
AGENT => Inventory successfully sent
	INVENTORY => Writing new inventory state
AGENT =>  Communication Server ask for Package Download
	DOWNLOAD => Package history file cleaning not required
	DOWNLOAD => Download disabled by server
AGENT => Unloading communication provider
AGENT => Unloading plug-in(s)
AGENT => Execution duration: 00:00:11.
```

**Comment by gillesdubois** [comment]
Hi,

You should check if your files are correctly installed in the default plugin directory (/etc/ocsinventory-server/*)
You should have a Map.pm file and a *.conf file.
After that try to restart your apache service.

Regards,
Gilles Dubois.

**Comment by JulienG78** [comment]
Hi, I have the same problem.

I have installed all plugins, I see them on the server.
My Windows agent has the plugins vbs files but nothing comes up with the inventory.

I see in my etc/ocsinventory-server/plugins folder, it's empty.
Alors in the /usr/share/ocsinventory-reports/ocsreports I see the default plugins but not the others I have installed.
I think it might be the reason why its not working, but I don't know what to do because it looks like its installed, on the webinterface in the plugins manager they are installed, and in the machine Miscellaneous tab I see the plugins fields, but they are empty.

So what files do I have to put and where ?

Thank you.
